From 1621f6271bad1e309d020cb5f0209133aa7bece9 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Thu, 30 Apr 2015 08:22:41 +0200
Subject: [PATCH 1514/2508] ALSA: hda - Reduce ifdef CONFIG_SND_HDA_I915

commit 31c648e0d304bbfc49a4a22c6139718367f3fd8d upstream

We're providing dummy functions for CONFIG_SND_HDA_i915=n, thus ifdef
can be reduced.  (But hda_i915_init() has to be fixed to return zero.)

This automatically fixes a compile warning:
  sound/pci/hda/hda_intel.c: In function 'azx_probe_continue':
  sound/pci/hda/hda_intel.c:1966:1: warning: label 'i915_power_fail' defined but not used [-Wunused-label]

Reported-by: kbuild test robot <fengguang.wu@intel.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 sound/pci/hda/hda_intel.c |    3 ---
 sound/pci/hda/hda_intel.h |    2 +-
 2 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/sound/pci/hda/hda_intel.c b/sound/pci/hda/hda_intel.c
index 5edd2370..5dc00f8 100644
--- a/sound/pci/hda/hda_intel.c
+++ b/sound/pci/hda/hda_intel.c
@@ -1951,8 +1951,6 @@ static int azx_probe_continue(struct azx *chip)
 		if (pci->device != 0x0f04 && pci->device != 0x2284)
 			hda->need_i915_power = 1;
 
-
-#ifdef CONFIG_SND_HDA_I915
 		err = hda_i915_init(hda);
 		if (err < 0)
 			goto i915_power_fail;
@@ -1963,7 +1961,6 @@ static int azx_probe_continue(struct azx *chip)
 				"Cannot turn on display power on i915\n");
 			goto i915_power_fail;
 		}
-#endif
 	}
 
 #ifdef CONFIG_SND_HDA_I915
diff --git a/sound/pci/hda/hda_intel.h b/sound/pci/hda/hda_intel.h
index 505f987..ff41fc30 100644
--- a/sound/pci/hda/hda_intel.h
+++ b/sound/pci/hda/hda_intel.h
@@ -63,7 +63,7 @@ static inline int hda_display_power(struct hda_intel *hda, bool enable)
 static inline void haswell_set_bclk(struct hda_intel *hda) { return; }
 static inline int hda_i915_init(struct hda_intel *hda)
 {
-	return -ENODEV;
+	return 0;
 }
 static inline int hda_i915_exit(struct hda_intel *hda)
 {
-- 
1.7.5.4

